### PR TITLE
applications: nrf_desktop: Add click detector documentation

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds documentation for click detector module.

### CODE REVIEW

**Comment by MarekPieta** [comment]
@greg-fer, could you review ?

File: applications/nrf_desktop/doc/click_detector.rst [hunk 1]
```diff
@@ -1,6 +1,50 @@
...
+The ``click_detector`` module is used to send ``click_event`` when recognized type of click is recorded for button defined in the module configuration.
+
+Click type
+**********
+
+The module records the following click types:
+* :cpp:enumerator:`CLICK_SHORT`
+  Button pressed and released after short time.
+* :cpp:enumerator:`CLICK_NONE`
+  Button pressed and held for time too long for :cpp:enumerator:`CLICK_SHORT`, but too short for :cpp:enumerator:`CLICK_LONG`.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/click_detector.rst]:
```suggestion
 Button pressed and held for a period of time that is too long for :cpp:enumerator:`CLICK_SHORT`, but too short for :cpp:enumerator:`CLICK_LONG`.
```

File: applications/nrf_desktop/doc/click_detector.rst [hunk 2]
```diff
@@ -1,6 +1,50 @@
...
+Click detector module
 #####################
 
-The ``click_detector`` module tracks button events to detect clicks.
+The ``click_detector`` module is used to send ``click_event`` when recognized type of click is recorded for button defined in the module configuration.
+
+Click type
+**********
+
+The module records the following click types:
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/click_detector.rst]: Rework the list to the following formatting:
* :cpp:enumerator:`CLICK_SHORT` - Button pressed and released after short time.

File: applications/nrf_desktop/doc/click_detector.rst [hunk 3]
```diff
@@ -1,6 +1,50 @@
 .. _click_detector:
 
-click_detector module
+Click detector module
 #####################
 
-The ``click_detector`` module tracks button events to detect clicks.
+The ``click_detector`` module is used to send ``click_event`` when recognized type of click is recorded for button defined in the module configuration.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/click_detector.rst]:
```suggestion
The ``click_detector`` module is used to send a ``click_event`` when a known type of click is recorded for the button defined in the module configuration.
```

File: applications/nrf_desktop/doc/click_detector.rst [hunk 4]
```diff
@@ -1,6 +1,50 @@
...
+Click type
+**********
+
+The module records the following click types:
+* :cpp:enumerator:`CLICK_SHORT`
+  Button pressed and released after short time.
+* :cpp:enumerator:`CLICK_NONE`
+  Button pressed and held for time too long for :cpp:enumerator:`CLICK_SHORT`, but too short for :cpp:enumerator:`CLICK_LONG`.
+* :cpp:enumarator:`CLICK_LONG`
+  Button pressed and held for long time.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/click_detector.rst]:
```suggestion
 Button pressed and held for a long period of time.
```

File: applications/nrf_desktop/doc/click_detector.rst [hunk 5]
```diff
@@ -1,6 +1,50 @@
...
+* :cpp:enumerator:`CLICK_DOUBLE`
+  Two sequences of button press and release in short time interval.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/click_detector.rst]:
```suggestion
 Two sequences of the button press and release in a short time interval.
```

File: applications/nrf_desktop/doc/click_detector.rst [hunk 6]
```diff
@@ -1,6 +1,50 @@
...
+
+Exact values of time intervals for click types are defined in ``click_detector.c`` file.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/click_detector.rst]:
```suggestion
The exact values of time intervals for click types are defined in the ``click_detector.c`` file.
```

File: applications/nrf_desktop/doc/click_detector.rst [hunk 7]
```diff
@@ -1,6 +1,50 @@
...
+| Source Module  | Input Event      | This Module        | Output Event     | Sink Module      |
++================+==================+====================+==================+==================+
+| :ref:`buttons` | ``button_event`` | ``click_detector`` | ``click_event``  | :ref:`ble_bond`  |
++----------------+------------------+--------------------+------------------+------------------+
+
+Configuration
+*************
+
+Module detects click types based on ``button_event``.
+You should define :ref:`buttons` hardware interface.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/click_detector.rst]:
```suggestion
Make sure you define the :ref:`buttons` hardware interface.
```

File: applications/nrf_desktop/doc/click_detector.rst [hunk 8]
```diff
@@ -1,6 +1,50 @@
...
+
+Set ``CONFIG_DESKTOP_CLICK_DETECTOR_ENABLE`` and define module configuration.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/click_detector.rst]:
```suggestion
Set ``CONFIG_DESKTOP_CLICK_DETECTOR_ENABLE`` and define the module configuration.
```

File: applications/nrf_desktop/doc/click_detector.rst [hunk 9]
```diff
@@ -1,6 +1,50 @@
...
+The configuration (array of :c:type:`struct click_detector_config`) is written in the ``click_detector_def.h`` file located in the board-specific directory in the application configuration folder.
+
+For every click detector you should define:
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/click_detector.rst]:
```suggestion
For every click detector, you should define:
```

File: applications/nrf_desktop/doc/click_detector.rst [hunk 10]
```diff
@@ -1,6 +1,50 @@
...
+* :cpp:member:`key_id` - ID of selected key
+* :cpp:member:`consume_button_event` - define if the ``button_event`` with given :cpp:member:`key_id` should be consumed by the module
+
+Implementation details
+**********************
+
+Tracing of key states is implemented using periodically submitted work - :c:type:`struct k_delayed_work`.
+The work updates states of traced keys and sends ``click_event``.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/click_detector.rst]:
```suggestion
The work updates the states of traced keys and sends ``click_event``.
```

File: applications/nrf_desktop/doc/click_detector.rst [hunk 11]
```diff
@@ -1,6 +1,50 @@
...
+The work is not submitted if there is no key, which state should be updated.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/click_detector.rst]: This is not clear. What is the meaning?

File: applications/nrf_desktop/doc/click_detector.rst (hunk 2 above)
Comment (MarekPieta) [reply applications/nrf_desktop/doc/click_detector.rst]: done

File: applications/nrf_desktop/doc/click_detector.rst (hunk 11 above)
Comment (MarekPieta) [reply applications/nrf_desktop/doc/click_detector.rst]: Discussed offline

**Pull request merged**